client.go

```diff
@@ -1468,6 +1468,32 @@ func (cli *Client) JoinedRooms() (resp *RespJoinedRooms, err error) {
 	return
 }
 
+// Hierarchy returns a list of rooms that are in the room's hierarchy. See https://spec.matrix.org/v1.4/client-server-api/#get_matrixclientv1roomsroomidhierarchy
+//
+// The hierarchy API is provided to walk the space tree and discover the rooms with their aesthetic details. works in a depth-first manner:
+// when it encounters another space as a child it recurses into that space before returning non-space children.
+//
+// NOTE: Only the roomID is required, the rest of the fields are optional and ignored if left to default values.
+func (cli *Client) Hierarchy(roomID id.RoomID, from string, max_depth int, suggested_only bool, limit int) (resp *RespHierarchy, err error) {
```

Comment:
Go variables should be camelCase.

```suggestion
func (cli *Client) Hierarchy(roomID id.RoomID, from string, maxDepth int, suggestedOnly bool, limit int) (resp *RespHierarchy, err error) {
```